.changeset/breezy-garlics-bathe.md

```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/action": minor
```

Comment:
IMHO the proposed change is a major change. It's a substantial change in the default behavior. If anybody is relying on publish being an opt-in then it could have a devastating effect for them if we suddenly try to publish their packages out of a blue.

Reply:
Yeah, like I'm personally using it in places where publishing is handled separately where I definitely wouldn't want this behaviour and the README documents a `Without Publishing` workflow which with this PR isn't accurate

Reply:
Ah, thanks! I'd never used this without publishing, so sorry I missed that was a way to use this. I've updated this PR to revert to the previous behavior but update the error message

Reply:
We can take another look if this is he best default behavior but that will have to be done as part of the next major release